engine/common/rpc/convert/convert_test.go

```diff
@@ -30,12 +30,114 @@ func TestConvertAccountKey(t *testing.T) {
 	accountKey.Revoked = true
 
 	msg, err := convert.AccountKeyToMessage(accountKey)
-	assert.Nil(t, err)
+	assert.NoError(t, err)
 
 	converted, err := convert.MessageToAccountKey(msg)
-	assert.Nil(t, err)
+	assert.NoError(t, err)
 
 	assert.Equal(t, accountKey, *converted)
 	assert.Equal(t, accountKey.PublicKey, converted.PublicKey)
 	assert.Equal(t, accountKey.Revoked, converted.Revoked)
 }
+
+func TestConvertFullBlock(t *testing.T) {
+	fullBlock := unittest.FullBlockFixture()
+
+	msg, err := convert.BlockToMessage(&fullBlock)
+	assert.NoError(t, err)
+
+	// extract payload for payload hash check
+	extractedPayload, err := convert.PayloadFromMessage(msg)
+	assert.NoError(t, err)
+
+	converted, err := convert.MessageToBlock(msg)
+	assert.NoError(t, err)
+
+	assert.Equal(t, converted.ID(), fullBlock.ID())
+	assert.Equal(t, converted.Payload.Hash(), fullBlock.Payload.Hash())
+	assert.Equal(t, extractedPayload.Hash(), fullBlock.Payload.Hash())
+}
+
+func TestConvertBlockHeader(t *testing.T) {
+	header := unittest.BlockHeaderFixture()
+
+	msg, err := convert.BlockHeaderToMessage(&header)
+	assert.NoError(t, err)
+
+	converted, err := convert.MessageToBlockHeader(msg)
+	assert.NoError(t, err)
+
+	assert.Equal(t, converted.ID(), header.ID())
+}
+
+func TestConvertSeal(t *testing.T) {
+	blockSeals := unittest.BlockSealsFixture(3)
+
+	msg := convert.BlockSealsToMessages(blockSeals)
+
+	converted, err := convert.MessagesToBlockSeals(msg)
+	assert.NoError(t, err)
+
+	for idx, convertedSeal := range converted {
+		assert.Equal(t, convertedSeal.ID(), blockSeals[idx].ID())
+	}
+}
+
+func TestConvertEvent(t *testing.T) {
+	txId := unittest.TransactionFixture().ID()
+	setupEvent := unittest.EventFixture(flow.ServiceEventSetup, uint32(1), uint32(1), txId, 0)
+
+	msg := convert.EventToMessage(setupEvent)
+	converted := convert.MessageToEvent(msg)
+	assert.Equal(t, setupEvent.ID(), converted.ID())
+
+	commitEvent := unittest.EventFixture(flow.ServiceEventCommit, uint32(1), uint32(1), txId, 0)
+
+	msg = convert.EventToMessage(commitEvent)
+	converted = convert.MessageToEvent(msg)
+	assert.Equal(t, setupEvent.ID(), converted.ID())
```

Comment:
nit: is there any way to systematically iterate through all Event types so that when new ones are added they are automatically covered by this test? For example, maybe define a new function `ServiceEventTypes()` in `service_event.go` right underneath the service event definitions which returns a list of all existing service events, with a comment saying that it must be updated every time a new event is added?

Reply:
worth figuring out let's see if there's a golang way to iterate through it